src/primitives/transaction.h

```diff
@@ -200,6 +200,51 @@ class CTxOut
 extern boost::atomic<bool> flexTransActive;
 struct CMutableTransaction;
 
+template<typename Stream>
+void SerialiseScriptSig4(const std::vector<CTxIn> &inputs, Stream &s, int nType, int nVersion)
+{
+    for (auto in : inputs) {
+        bool first = true;
+        unsigned int i = 0; // use both iterator and int b/c the iterator doesn't do bounds-checks.
+        auto iter = in.scriptSig.begin();
+        while (i < in.scriptSig.size()) {
+            uint8_t k = *iter;
+            if ((k > 0 && k < 76) || (k >= 76 && k <= 78)) {
+                uint32_t size = k;
+                if (k >= 76) { // OP_PUSHDATA
+                    const int width = k - 75; // address width
+                    size = *(++iter);
+                    if (width > 1) {
+                        size = (size << 8) + (uint8_t) *(++iter);
+                        if (width > 2) {
+                            size = (size << 8) + (uint8_t) *(++iter);
+                            size = (size << 8) + (uint8_t) *(++iter);
+                            ++i;
+                        }
+                    }
+                    i += width;
+                }
+                if (size + i >= in.scriptSig.size())
+                    throw std::runtime_error("Signatures malformed");
+                auto iterBegin = ++iter;
+                iter += size;
+                CMFToken token(first ? Consensus::TxInputStackItem : Consensus::TxInputStackItemContinued,
+                    std::vector<char>(iterBegin, iter));
+                STORECMF(token);
+                i += size;
+            } else if (k == OP_FALSE) {
+                CMFToken token(first ? Consensus::TxInputStackItem : Consensus::TxInputStackItemContinued,
+                               std::vector<char>(1, OP_FALSE));
+                STORECMF(token);
+                ++iter;
+            } else
+                throw std::runtime_error("Signatures malformed");
+            i++;
+            first = false;
+        }
+    }
```

Comment:
i'm not super fan fof this approach, but I understand this allow to get something up and running quickly to test with.